lib/src/facts/linux/networking_resolver.cc

```diff
@@ -24,6 +24,35 @@ namespace facter { namespace facts { namespace linux {
         read_routing_table();
         data result = bsd::networking_resolver::collect_data(facts);
         populate_from_routing_table(result);
+
+        // On linux, the macaddress of bonded interfaces is reported
+        // as the address of the bonding master. We want to report the
+        // original HW address, so we dig it out of /proc
+        for (auto& interface : result.interfaces) {
+            // For each interface we check if we're part of a bond,
+            // and update the `macaddress` fact if we are
+            auto bond_master = get_bond_master(interface.name);
+            if (!bond_master.empty()) {
+                bool in_our_block = false;
+                lth_file::each_line("/proc/net/bonding/"+bond_master, [&](string& line) {
+                    // /proc/net/bonding files are organized into chunks for each slave
+                    // interface. We want to grab the mac address for the block we're in.
+                    if (line == "Slave Interface: " + interface.name) {
+                        in_our_block = true;
+                    } else if (line.find("Slave Interface") != string::npos) {
+                        in_our_block = false;
+                    }
+
+                    // If we're in the block for our iface, we can grab the HW address
+                    if (in_our_block && line.find("Permanent HW addr: ") != string::npos) {
+                        auto split = line.find(':') + 2;
+                        interface.macaddress = line.substr(split, string::npos);
```

Comment:
I guess I could use the boost tokenizer, rather than searching the string myself. We're already using that throughout facter